numpy/f2py/tests/test_regression.py

```diff
@@ -27,3 +27,17 @@ def test_inout(self):
         x = np.arange(3, dtype=np.float32)
         self.module.foo(x)
         assert_equal(x, [3, 1, 2])
+
+@pytest.mark.parametrize('code', [
+        'program test_f2py\nend program test_f2py',
+        b'program test_f2py\nend program test_f2py',
+    ])
+def test_compile(tmpdir, code):
```

Comment:
Where does `tmpdir` come from?

Reply:
It is a [pytest fixture](https://docs.pytest.org/en/latest/tmpdir.html), automatically provided and removed when the test is over

Reply:
Does it need an import?

Reply:
Seems to be failing on Python 3.5.

Reply:
passes now, thanks